ui/components/multichain/select-action-modal/select-action-modal.js

```diff
@@ -0,0 +1,249 @@
+  MetaMetricsSwapsEventSource,
+  ///: END:ONLY_INCLUDE_IN
+} from '../../../../shared/constants/metametrics';
+import {
+  getCurrentChainId,
+  getIsSwapsChain,
+  ///: BEGIN:ONLY_INCLUDE_IN(build-main,build-beta,build-flask)
+  getSwapsDefaultToken,
+  getCurrentKeyring,
+  getIsBridgeChain,
+  getIsBuyableChain,
+  getMetaMetricsId,
+  ///: END:ONLY_INCLUDE_IN
+} from '../../../selectors';
+import {
+  ///: BEGIN:ONLY_INCLUDE_IN(build-main,build-beta,build-flask)
+  BUILD_QUOTE_ROUTE,
+  ///: END:ONLY_INCLUDE_IN
+  SEND_ROUTE,
+} from '../../../helpers/constants/routes';
+import { startNewDraftTransaction } from '../../../ducks/send';
+import { I18nContext } from '../../../contexts/i18n';
+import { AssetType } from '../../../../shared/constants/transaction';
+///: BEGIN:ONLY_INCLUDE_IN(build-mmi)
+import { MMI_SWAPS_URL } from '../../../../shared/constants/swaps';
+import { MMI_STAKE_WEBSITE } from '../../../helpers/constants/common';
+///: END:ONLY_INCLUDE_IN
+///: BEGIN:ONLY_INCLUDE_IN(build-main,build-beta,build-flask)
+import { setSwapsFromToken } from '../../../ducks/swaps/swaps';
+import { isHardwareKeyring } from '../../../helpers/utils/hardware';
+///: END:ONLY_INCLUDE_IN
+
+export const SelectActionModal = ({ onClose }) => {
+  const dispatch = useDispatch();
+  const t = useContext(I18nContext);
+  const trackEvent = useContext(MetaMetricsContext);
+  const history = useHistory();
+  const chainId = useSelector(getCurrentChainId);
+  const isSwapsChain = useSelector(getIsSwapsChain);
+
+  ///: BEGIN:ONLY_INCLUDE_IN(build-main,build-beta,build-flask)
+  const location = useLocation();
+  const { openBuyCryptoInPdapp } = useRamps();
+  const defaultSwapsToken = useSelector(getSwapsDefaultToken);
+  const keyring = useSelector(getCurrentKeyring);
+  const usingHardwareWallet = isHardwareKeyring(keyring?.type);
+  const isBridgeChain = useSelector(getIsBridgeChain);
+  const metaMetricsId = useSelector(getMetaMetricsId);
+  const isBuyableChain = useSelector(getIsBuyableChain);
+
+  ///: END:ONLY_INCLUDE_IN
+
+  ///: BEGIN:ONLY_INCLUDE_IN(build-mmi)
+  const stakingEvent = () => {
+    trackEvent({
+      category: MetaMetricsEventCategory.Navigation,
+      event: MetaMetricsEventName.MMIPortfolioButtonClicked,
+    });
+  };
+  ///: END:ONLY_INCLUDE_IN
+
+  return (
+    <Modal
+      isOpen
+      onClose={onClose}
+      className="select-action-modal"
+      data-testid="select-action-modal"
+    >
+      <ModalOverlay />
+      <ModalContent>
+        <ModalHeader onClose={onClose}>{t('selectAnAction')}</ModalHeader>
+        <Box className="select-action-modal__container" marginTop={6}>
+          {
+            ///: BEGIN:ONLY_INCLUDE_IN(build-main,build-beta,build-flask)
+            <SelectActionModalItem
+              actionIcon={IconName.Add}
+              showIcon
+              primaryText={t('buy')}
+              secondaryText={t('buyDescription')}
+              isDisabled={!isBuyableChain}
```

Comment:
All of our other components just use the `disabled` prop -- I don't think we need the `is`